File: output/deepseek_chat/avatar/algo_based_translation/python/java/successful_translation/atcoder_ABC172_D.java
import java.util.*;
import java.io.*;

public class atcoder_ABC172_D {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine().trim());
        long ans = 0;
        
        for (int i = 1; i <= N; i++) {
            int quotient = N / i;
            long term = (long) i * quotient * (quotient + 1) / 2;
            ans += term;
        }
        
        System.out.println(ans);
    }
}

File: output/deepseek_chat/avatar/algo_based_translation/java/python/atcoder_ABC172_D.py

import sys

def main():
    n = int(sys.stdin.readline())
    ans = 0
    for i in range(1, n + 1):
        term = i * (n // i) * ((n // i) + 1) // 2
        ans += term
    print(ans)

if __name__ == "__main__":
    main()

